tests/SentryMiddlewareTest.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Sentry\Tests;
+
+use HttpSoft\Message\Response;
+use HttpSoft\Message\ServerRequest;
+use PHPUnit\Framework\Error\Error;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use RuntimeException;
+use Yiisoft\Yii\Sentry\SentryMiddleware;
+use Yiisoft\Yii\Sentry\Tests\Stub\Transport;
+
+final class SentryMiddlewareTest extends TestCase
+{
+    public function testProcessWithException(): void
+    {
+        $methodName = debug_backtrace()[0]['function'];
+        $eventKey = self::class . "::$methodName()";
+
+        $this->expectException(RuntimeException::class);
+        $this->expectExceptionMessage('Exception test.');
+
+        try {
+            $middleware = new SentryMiddleware($this->createSentryHub($eventKey));
+            $middleware->process(
+                new ServerRequest(method: 'GET', uri: '/'),
+                $this->createRequestHandlerWithException(),
+            );
+        } catch (RuntimeException $e) {
+            $this->assertTransportHasException('RuntimeException', 'Exception test.', $eventKey);
+
+            throw $e;
```

Comment:
What's the reason to throw it?

Reply:
Because of performing additional assertions after exception has been thrown. Otherwise either this won't be possible or `exceptException()` and `exceptExceptionMessage()` won't be working correctly. The solution is based on this StackOverflow question and answer - https://stackoverflow.com/a/53324493/4323648. Maybe there is a simpler way to do it, let me know in this case.